### PR TITLE
Fix DDP download problem

### PR DESCRIPTION
**Pull request by DifferentSC**
Fix multiple download problems when running on DDP.

### CODE REVIEW

File: src/periflow_sdk/examples/cifar10/main.py [hunk 1]
```diff
@@ -68,13 +68,12 @@ class CIFAR10TrainStepOutput(TrainStepOutput):
     torch.cuda.set_device(args.local_rank)
     net.cuda(args.local_rank)
     net = torch.nn.parallel.DistributedDataParallel(net, device_ids=[args.local_rank])
+    is_download = args.local_rank
```
Comment (kooyunmo) [inline src/periflow_sdk/examples/cifar10/main.py:71]: useless line?

**Review by kooyunmo — CHANGES_REQUESTED** [review]
Please merge this PR after resolving one minor comment.

**Comment by kooyunmo** [comment]
When I run the script on this PR, it raises the following error

```sh
Traceback (most recent call last):
   File "main.py", line 84, in <module>
     sampler = ResumableRandomSampler(len(trainset),
 NameError: name 'trainset' is not defined
```

```python
is_ddp = world_size > 1

if is_ddp:
    torch.distributed.init_process_group(backend='nccl', init_method='env://')
    torch.cuda.set_device(args.local_rank)
    net.cuda(args.local_rank)
    net = torch.nn.parallel.DistributedDataParallel(net, device_ids=[args.local_rank])
    download = args.local_rank == 0

trainset = torchvision.datasets.CIFAR10(
    root='./data', train=True, download=download, transform=transform_train)

elif device == 'cuda':
    net = torch.nn.DataParallel(net)
    cudnn.benchmark = True

if is_ddp:
    # Use distributed sampler
    sampler = ResumableRandomSampler(len(trainset),
                                     args.batch_size,
                                     False,
                                     77,
                                     args.local_rank,
                                     world_size)
else:
    sampler = ResumableRandomSampler(len(trainset), args.batch_size, False)
    download = True

trainloader = torch.utils.data.DataLoader(
    trainset, batch_sampler=sampler, num_workers=4)

testset = torchvision.datasets.CIFAR10(
    root='./data', train=False, download=True, transform=transform_test)
testloader = torch.utils.data.DataLoader(
    testset, batch_size=100, shuffle=False, num_workers=4)
```

**Comment by kooyunmo** [comment]
BTW, downloading the dataset only at the local rank 0 still raises the following error:

```sh
00:55:37 [168.62.51.139]: [36mperiflow-0           |[0m Traceback (most recent call last):
00:55:37 [168.62.51.139]: [36mperiflow-0           |[0m   File "main.py", line 83, in <module>
00:55:37 [168.62.51.139]: [36mperiflow-0           |[0m     trainset = torchvision.datasets.CIFAR10(
00:55:37 [168.62.51.139]: [36mperiflow-0           |[0m   File "/opt/conda/lib/python3.8/site-packages/torchvision/datasets/cifar.py", line 68, in __init__
00:55:37 [168.62.51.139]: [36mperiflow-0           |[0m     raise RuntimeError('Dataset not found or corrupted.' +
00:55:37 [168.62.51.139]: [36mperiflow-0           |[0m RuntimeError: Dataset not found or corrupted. You can use download=True to download it
00:55:37 [168.62.51.139]: [36mperiflow-0           |[0m Traceback (most recent call last):
00:55:37 [168.62.51.139]: [36mperiflow-0           |[0m   File "main.py", line 83, in <module>
00:55:37 [168.62.51.139]: [36mperiflow-0           |[0m     trainset = torchvision.datasets.CIFAR10(
00:55:37 [168.62.51.139]: [36mperiflow-0           |[0m   File "/opt/conda/lib/python3.8/site-packages/torchvision/datasets/cifar.py", line 68, in __init__
00:55:37 [168.62.51.139]: [36mperiflow-0           |[0m     raise RuntimeError('Dataset not found or corrupted.' +
00:55:37 [168.62.51.139]: [36mperiflow-0           |[0m RuntimeError: Dataset not found or corrupted. You can use download=True to download it
00:55:37 [168.62.51.139]: [36mperiflow-0           |[0m Traceback (most recent call last):
00:55:37 [168.62.51.139]: [36mperiflow-0           |[0m   File "main.py", line 83, in <module>
00:55:37 [168.62.51.139]: [36mperiflow-0           |[0m     trainset = torchvision.datasets.CIFAR10(
00:55:37 [168.62.51.139]: [36mperiflow-0           |[0m   File "/opt/conda/lib/python3.8/site-packages/torchvision/datasets/cifar.py", line 68, in __init__
00:55:37 [168.62.51.139]: [36mperiflow-0           |[0m     raise RuntimeError('Dataset not found or corrupted.' +
00:55:37 [168.62.51.139]: [36mperiflow-0           |[0m RuntimeError: Dataset not found or corrupted. You can use download=True to download it
00:55:37 [168.62.51.139]: [36mperiflow-0           |[0m Traceback (most recent call last):
00:55:37 [168.62.51.139]: [36mperiflow-0           |[0m   File "main.py", line 83, in <module>
00:55:37 [168.62.51.139]: [36mperiflow-0           |[0m     trainset = torchvision.datasets.CIFAR10(
00:55:37 [168.62.51.139]: [36mperiflow-0           |[0m   File "/opt/conda/lib/python3.8/site-packages/torchvision/datasets/cifar.py", line 68, in __init__
00:55:37 [168.62.51.139]: [36mperiflow-0           |[0m     raise RuntimeError('Dataset not found or corrupted.' +
00:55:37 [168.62.51.139]: [36mperiflow-0           |[0m RuntimeError: Dataset not found or corrupted. You can use download=True to download it
00:55:37 [168.62.51.139]: [36mperiflow-0           |[0m Traceback (most recent call last):
00:55:37 [168.62.51.139]: [36mperiflow-0           |[0m Traceback (most recent call last):
00:55:37 [168.62.51.139]: [36mperiflow-0           |[0m   File "main.py", line 83, in <module>
00:55:37 [168.62.51.139]: [36mperiflow-0           |[0m     trainset = torchvision.datasets.CIFAR10(
00:55:37 [168.62.51.139]: [36mperiflow-0           |[0m   File "main.py", line 83, in <module>
00:55:37 [168.62.51.139]: [36mperiflow-0           |[0m     trainset = torchvision.datasets.CIFAR10(
00:55:37 [168.62.51.139]: [36mperiflow-0           |[0m   File "/opt/conda/lib/python3.8/site-packages/torchvision/datasets/cifar.py", line 68, in __init__
00:55:37 [168.62.51.139]: [36mperiflow-0           |[0m     raise RuntimeError('Dataset not found or corrupted.' +
00:55:37 [168.62.51.139]: [36mperiflow-0           |[0m Traceback (most recent call last):
00:55:37 [168.62.51.139]: [36mperiflow-0           |[0m RuntimeError: Dataset not found or corrupted. You can use download=True to download it
00:55:37 [168.62.51.139]: [36mperiflow-0           |[0m   File "/opt/conda/lib/python3.8/site-packages/torchvision/datasets/cifar.py", line 68, in __init__
00:55:37 [168.62.51.139]: [36mperiflow-0           |[0m     raise RuntimeError('Dataset not found or corrupted.' +
00:55:37 [168.62.51.139]: [36mperiflow-0           |[0m RuntimeError: Dataset not found or corrupted. You can use download=True to download it
00:55:37 [168.62.51.139]: [36mperiflow-0           |[0m   File "main.py", line 83, in <module>
00:55:37 [168.62.51.139]: [36mperiflow-0           |[0m     trainset = torchvision.datasets.CIFAR10(
00:55:37 [168.62.51.139]: [36mperiflow-0           |[0m   File "/opt/conda/lib/python3.8/site-packages/torchvision/datasets/cifar.py", line 68, in __init__
00:55:37 [168.62.51.139]: [36mperiflow-0           |[0m     raise RuntimeError('Dataset not found or corrupted.' +
00:55:37 [168.62.51.139]: [36mperiflow-0           |[0m RuntimeError: Dataset not found or corrupted. You can use download=True to download it
00:55:37 [168.62.51.139]: [36mperiflow-0           |[0m Downloading https://www.cs.toronto.edu/~kriz/cifar-10-python.tar.gz to ./data/cifar-10-python.tar.gz
```

**Comment by DifferentSC** [comment]
Addressed in #13

**Pull request closed without merging**